### ISSUE TITLE
Bundling additional resources creates separate directory for each

### ISSUE THREAD

**Issue by jatin69** [brentvollebregt/auto-py-to-exe#4]
When I bundle up the additional resource. It autocompletes the destination part with the file name itself, and then what actually happens is, it creates a separate folder for each asset.  
![bugfix2](https://user-images.githubusercontent.com/13359891/37509495-93b4c54e-291d-11e8-8491-d3750ad928eb.PNG)

That's counter-intuitive. While developing the app it is a fairly common practice to bundle them up in one folder.  But here, we get a different folder for each, then I have to manually edit all the bundled resources' destination to reflect a single directory. 
![bugfix3](https://user-images.githubusercontent.com/13359891/37509508-9e57e27e-291d-11e8-9591-b4accd32734e.PNG)

My suggestion is instead of making a separate destination, just make a `asset` folder and place everything inside it.

Although far-fetched, but it'll be of great help, if we could upload all assets at once, instead of individually doing them. Nevertheless, that's an enhancement.

**Comment by brentvollebregt** [comment]
The whole idea of the additional files autocomplete is to give an option of what to call your file; you still need to edit them if you want it to be something else, you also need to make sure the data you are providing the converter is accurate. Also you can't just add a whole folder, you need to declare each file; this is standard pyinstaller practice.

The files being generated where you don't want them is not something I can control, this interface simply passes the information you gave it to pyinstaller and then does some cleaning up.

**Comment by jatin69** [comment]
1. The whole point of making a wrapper is to ease the work. So taking in a directory, seeing all its files, adding them individually with a loop, is possible I think. It obviously needs time, which you might not have, that's understandable. 
2. The passing is done in such a way, it makes a separate folder for each. If we pass same folder name to pyinstaller, all of them are in one directory. I made this work in my case, and can definitely be done here.
But again, time 😆 
Anyway, nice work on the app. 👏 👏

**Comment by brentvollebregt** [comment]
I see what you mean by the first option and yes time is definitely a factor but it is a good idea.

I don't quite get what you mean in two though. How would you generally pass the folder in pyinstaller? I didn't think it was possible.

**Comment by jatin69** [comment]
I'll try to explain it. see.

# What's happening
See this image, this is produced when I simply add an asset. I provide it a path via explorer, then it automatically sets the value of destination same as file name. 

![bugfix2](https://user-images.githubusercontent.com/13359891/37509495-93b4c54e-291d-11e8-8491-d3750ad928eb.PNG)

As a result, the command generated for pyinstaller is something of the format,
`"D:/mypath/assets/file1.png";"file1.png"`
`"D:/mypath/assets/file2.png";"file2.png"`
The string after the semicolon is for specifying the folder, where the file will live, once it has been bundled by pyinstaller `[--add-data <SRC;DEST or SRC:DEST>] `. 
As a result, if I specify 10 assets, each of them will live in a separate folder.
**As a result, the code fails, because the asset paths get messed up**, from what I was initially using.
As a user simply using your tool, I might not know that this will happen, so I can't specify the expected path in my code beforehand. 

# What I am suggesting
Simply ask the user for the asset file, fetch it, and autofill the `destination` text field with value `asset` no with its filename. No matter how many files the user adds, the default textbox value will be `asset`. This will ensure that all the asset files go to a folder name `asset` in the bundle.
You can ask the user to specify the path to all resources in his code in this format. `asset/file.png`.
Any experienced user will further subfolders in the asset folder will know what to change and what to expect. But for first-timers, this can be overwhelming to debug.  It is cumbersome, but at least it works. In the above approach, the code just crashes.

![bugfix3](https://user-images.githubusercontent.com/13359891/37509508-9e57e27e-291d-11e8-9591-b4accd32734e.PNG)

## Conculsion 
For the destination field, make the default value as `assets` unless user wants to change it.
Experienced user can tailor this folder schema to their needs. New ones, can simply follow.
Direct the user to place his assets in one folder, and use path to that folder in this python script, while using assets. It'll be also nice to tell the user about the [Temp issue and its solution from stackoverflow](https://stackoverflow.com/a/13790741/5371996) when using one file bundled mode with assets.
It'll surely save time for many.

Hope this helps.

**Comment by brentvollebregt** [comment]
Oh now I see, I swear this used to be different in older versions. I am very sorry, I was expecting DEST to be the filename/path e.g. "file.jpg" will create the file as "file.jpg" and assets/file.jpg will put it in the assets file named "file.jpg". Thank you for helping me realise this.

This I can fix quite quickly.

Also putting the temp issue and its solution from stackoverflow in is a great idea. Just not 100% on where I should put it. Just the README?

**Comment by brentvollebregt** [comment]
I believe this also fixed this the folder issue, unless you were thinking of making a whole folder of items become one item per additional file row.

**Comment by jatin69** [comment]
That's it. This commit fixes it. 👍 

I just realized one more thing, it's chrome. It cache's the code somehow. I had to inspect and disable cache to make this update work, the first time. Although, it won't bug the new users from here on. 
But I was wondering if there was a way in code to ensure `no-cache`, just in case if I ever need to disable cache somewhere. Any thoughts?

Just put a small line in the expansion of `Additional Files`, when the user clicks to expand this, A small line is there.
> Be careful when using additional files, update your code to work with pyinstaller, read [this](https://stackoverflow.com/questions/7674790/bundling-data-files-with-pyinstaller-onefile/13790741#13790741)
